### ISSUE TITLE
Launch PS Scripts using VB Script

### ISSUE THREAD

**Issue by samaguire** [MikeCarlo/BusinessOps#3]
You should consider launching the PS scripts using VB Script. There are several advantages to this:

- You can launch the PS script in an elevated mode if required
- There is no PS script window shown (no flashing)

See the attached example of this which I wrote. NB, the "runas" can be removed if window elevation isn't required.

[practicedataset.pbitool.json.zip](https://github.com/MikeCarlo/BusinessOps/files/5394625/practicedataset.pbitool.json.zip)

**Comment by MikeCarlo** [comment]
@JamesDBarlett3 Wondering your thoughts on this. Seems like a solid idea.

**Comment by samaguire** [comment]
You might want to have a look at my latest version: https://github.com/samaguire/PBIdesktop-PracticeDataset/releases/tag/v3.0.1

It is entirely PS driven, launching in bypass mode to get around security requirements and pulls the required files from GitHub if needed or a newer version is available.

**Comment by JamesDBartlett3** [comment]
@MikeCarlo This is indeed an interesting idea. If you open up the [practicedataset.pbitool.json](https://github.com/samaguire/PBIdesktop-PracticeDataset/releases/download/v3.0.1/practicedataset.pbitool.json) from @samaguire's previous comment, you'll see that the PowerShell command executes a ScriptBlock encoded in Base64. Here's what that ScriptBlock looks like in string format: 

![image](https://user-images.githubusercontent.com/37491308/162492425-be531574-22be-4309-8b74-8afb52d393f1.png)
It's actually fetching the latest version of everything it needs from the GitHub repo in real-time, stashing it on the hard drive, and then opening it from there. 

I see only one potential downside to this technique in its current implementation: The PowerShell script is hard-coded in the actual pbitool.json file itself, which is not one of the files auto-updated by the script, so any updates to the pbitool.json file (or the PowerShell script embedded therein) would still need to be performed manually. All of that being said, this kind of thing would definitely be a step in the right direction, particularly as we begin exploring ways to decouple the various external tool installers from the Business Ops installer.

**Comment by samaguire** [comment]
I agree with you @JamesDBartlett3. Embedding the logic in the pbitool.json isn't the most dynamic way to handle this.

I had thought about using this technique to pull down just a PS script that was the actual launcher to pull down other additional files, this way the pbitool.json could remain static. However, at the time I thought it was a bit overkill - I was in control of the GIT repo and any updates would be to the files that are being retrieved.
